src/commands.ts

```diff
@@ -162,7 +162,7 @@ const { commandKeyIndex, commandIdIndex, commandGestureIndex } = index()
 
 /** Gets the canonical gesture of the command as a string, ignoring aliases. Returns an empty string if the command does not have a gesture. */
 export const gestureString = (command: Command): string =>
-  (typeof command.gesture === 'string' ? command.gesture : command.gesture?.[0] || '') as string
+  (typeof command.gesture === 'string' ? command.gesture : command.gesture?.[0] || ' ') as string
```

Comment:
The reason for changing `''` to `' '` is because js evaluates expressions like `'rd'.startsWith('')` as true. Since cancel command's gesture is undefined `gestureString` will evaluate it as a empty string `''`. Due to this the comparison in `src/components/CommandPalette.tsx` will evaluate true for both cancel and the matched command gesture at the same time and they are selected simultaneously

Reply:
Thanks for the great explanation :). 

Let's either add this as a comment, or shift the responsibility to the CommandPalette. Otherwise this may not be obvious to other developers that look at this code.